element-connector/src/main/java/org/eclipse/californium/elements/tcp/TcpServerConnector.java

```diff
@@ -82,17 +85,17 @@ public TcpServerConnector(InetSocketAddress localAddress, int numberOfThreads, i
 				.option(ChannelOption.AUTO_READ, true).childHandler(new ChannelRegistry());
 
 		// Start the server.
-		bootstrap.bind(localAddress).syncUninterruptibly().addListener(new ChannelFutureListener() {
-
-			@Override
-			public void operationComplete(ChannelFuture future) throws Exception {
-				InetSocketAddress listenAddress = (InetSocketAddress) future.channel().localAddress();
-				listenUri = getListenUri(listenAddress);
-			}
-		});
+		ChannelFuture channelFuture = bootstrap.bind(localAddress).syncUninterruptibly();
+
+		if (channelFuture.isSuccess() && 0 == localAddress.getPort()) {
+			InetSocketAddress listenAddress = (InetSocketAddress) channelFuture.channel().localAddress();
+			localAddress = new InetSocketAddress(localAddress.getAddress(), listenAddress.getPort());
```

Comment:
Hm, in this PR I would like to got for "the conservative solution" and just replace the port.
I'm afraid, that changing towards an IPv6 address would introduce more problems outsite.
Additionally we may add this InetSocketAddress to URI in a common util class.